encord/objects/ontology_labels_impl.py

```diff
@@ -909,12 +914,28 @@ def data_type(self) -> DataType:
 
     @property
     def label_status(self) -> LabelStatus:
+        if self.__is_tms2_project:
+            raise WrongProjectTypeError(
+                '"label"_status property returns incorrect results for TMS2 projects.\
+             Please use "workflow_graph_node" property instead.'
+            )
+
         return self._label_row_read_only_data.label_status
 
     @property
     def annotation_task_status(self) -> AnnotationTaskStatus:
+        if self.__is_tms2_project:
+            raise WrongProjectTypeError(
+                '"annotation_task_status" property returns incorrect results for TMS2 projects.\
+             Please use "workflow_graph_node" property instead.'
+            )
+
         return self._label_row_read_only_data.annotation_task_status
 
+    @property
+    def workflow_graph_node(self) -> Optional[WorkflowGraphNode]:
```

Comment:
should we have an error for `if not self.__is_tms2_project`?

Reply:
This property returns Optional, and for non-tms2 projects there will be None, so the exception here is not strictly required. Null-check is enough. We could add it for the uniformity though, don't have too strong opinion here